test/Npgsql.Tests/Types/NumericTypeTests.cs

```diff
@@ -335,6 +329,59 @@ public void Numeric()
+                //midpoint rounding is AwayFromZero instead of ToEven https://msdn.microsoft.com/en-us/library/system.midpointrounding.aspx
+                //This is necessary to match psql.
+                var rounding = new Tuple<string, decimal>[] {
+                    new Tuple<string,decimal>("'1'::numeric(10,2)", 1.00M),     //test scale
+                    new Tuple<string,decimal>("'0'::numeric(10,2)", 0.00M),     //test scale
+                    new Tuple<string,decimal>("'1.2'::numeric(10,2)", 1.20M),   //test scale
+                    new Tuple<string,decimal>("'1.4'::numeric(10,0)", 1M),      //round down
+                    new Tuple<string,decimal>("'1.5'::numeric(10,0)", 2M),      //round up
+                    new Tuple<string,decimal>("'2.4'::numeric(10,0)", 2M),      //round down
+                    new Tuple<string,decimal>("'2.5'::numeric(10,0)", 3M),      //round up
+                    new Tuple<string,decimal>("'-1.4'::numeric(10,0)", -1M),    //round down
+                    new Tuple<string,decimal>("'-1.5'::numeric(10,0)", -2M),    //round up
+                    new Tuple<string,decimal>("'-2.4'::numeric(10,0)", -2M),    //round down
+                    new Tuple<string,decimal>("'-2.5'::numeric(10,0)", -3M),    //round up
+                };
+
+                foreach (var t in rounding)
+                {
+                    using (var cmd = new NpgsqlCommand($"SELECT {t.Item1}", conn))
+                    {
+                        var result = cmd.ExecuteScalar();
+                        Assert.AreEqual(decimal.GetBits(t.Item2), decimal.GetBits((decimal)result));
+                    }
+                }
+            }
+        }
+
+        //NOTE: it is important to test rounding on both even and odd numbers to make sure
+        //midpoint rounding is AwayFromZero instead of ToEven https://msdn.microsoft.com/en-us/library/system.midpointrounding.aspx
+        //This is necessary to match psql.
+        [Test]
+        [TestCase("'1'::numeric(10,2)", "1.00", TestName = "ExpandScaleOdd")]
+        [TestCase("'2'::numeric(10,2)", "2.00", TestName = "ExpandScaleEven")]
+        [TestCase("'1.2'::numeric(10,2)", "1.20", TestName = "ExpandScaleDecimal")]
+        [TestCase("'1.4'::numeric(10,0)", "1", TestName = "RoundOddDown")]
+        [TestCase("'1.5'::numeric(10,0)", "2", TestName = "RoundOddUp")]
+        [TestCase("'2.4'::numeric(10,0)", "2", TestName = "RoundEvenDown")]
+        [TestCase("'2.5'::numeric(10,0)", "3", TestName = "RoundEvenUp")]
+        [TestCase("'-1.4'::numeric(10,0)", "-1", TestName = "RoundNegativeOddDown")]
+        [TestCase("'-1.5'::numeric(10,0)", "-2", TestName = "RoundNegativeOddUp")]
+        [TestCase("'-2.4'::numeric(10,0)", "-2", TestName = "RoundNegativeEvenDown")]
+        [TestCase("'-2.5'::numeric(10,0)", "-3", TestName = "RoundNegativeEvenUp")]
+        public void NumericScaleCasting(string sql, string expected)
```

Comment:
Any reason for expected to be a string instead of a decimal directly?

Reply:
yup. decimal is not a constant type and can't be used where a constant expression is required.

Reply:
How do you mean? You can have decimal literals with `m`, and the compiler seems to accept `[TestCase(8m)]` just fine...

Reply:
@roji Nope, it will cause CS0182: An attribute argument must be a constant expression, typeof expression or array creation expression of an attribute parameter type. It's not a primitive type.

Reply:
Sorry about that, no red squgglies in Rider for some reason but you're right that it doesn't compile. Here's [an article](https://codeblog.jonskeet.uk/2014/08/22/when-is-a-constant-not-a-constant-when-its-a-decimal/) by Jon Skeet that goes a bit deep into this.